### DISCUSSION TITLE
Use "pre-compiler constants" to indetify as "USB" and not as "Virtual" interface?

### DISCUSSION THREAD

**Discussion by varoudis** [ExistentialAudio/BlackHole#770] [General]
Some apps will not work with "Virtual" interfaces and require their master out to be controlled from the central MacOS Settings->Sound and by setting "Output" to a USB device.

Is it possible to compile BalckHole to look like a USB interface?

**Comment by devinroth** [comment]
Absolutely.

I’ll add this to the list. In fact, I might just change it to USB by default because of that fact.

**Reply by gchilds** [reply]
I've wondered about "built-in" or even a "depends who's asking" response.

**Reply by varoudis** [reply]
What do I need to change to build it as USB?

**Reply by gchilds** [reply]
In this line 

https://github.com/ExistentialAudio/BlackHole/blob/master/BlackHole/BlackHole.c#L2543

change `kAudioDeviceTransportTypeVirtual` to `kAudioDeviceTransportTypeUSB`

The constant appears in another place, as `kAudioBoxPropertyTransportType`. I have _absolutely_ no idea what a box is or if it will affect your USBness.

**Comment by varoudis** [comment]
I can confirm that changing to "kAudioDeviceTransportTypeUSB" in two lines in the code did the trick!

Worked like a charm. In my case, I also removed 44.1 and below my output app is a mess and changes from 48 to 44.1 without asking...
All ok.
